pkg/services/ngalert/state/state_tracker.go

```diff
@@ -111,11 +114,11 @@ func (st *StateTracker) ResetCache() {
 	st.stateCache.cacheMap = make(map[string]AlertState)
 }
 
-func (st *StateTracker) ProcessEvalResults(alertRule *ngModels.AlertRule, results eval.Results) []AlertState {
+func (st *StateTracker) ProcessEvalResults(alertRule *ngModels.AlertRule, results eval.Results, processingTime time.Duration) []AlertState {
```

Comment:
We can and should. I chose this route to minimize the changes in this PR. `ConditionEval` is called enough places that I felt it would be more expedient to change the eval.Result struct to add this after we've ripped out all of the legacy code that's callin `ConditionEval`